src/main/java/org/quiltmc/loader/api/config/Config.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.loader.api.config;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.function.Consumer;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.loader.api.config.values.ValueList;
+import org.quiltmc.loader.api.config.values.ValueMap;
+import org.quiltmc.loader.api.config.values.ValueTreeNode;
+import org.quiltmc.loader.impl.config.ConfigImpl;
+
+@ApiStatus.NonExtendable
+public interface Config {
+	String getModId();
+
+	String getId();
+
+	/**
+	 * The path this config will be saved in, relative to Quilt Loader's config directory.
```

Comment:
since we're going to make this separate from quilt loader
```suggestion
	 * The path this config will be saved in, relative to the root config directory.
```